src/rules/no_const_assign.rs

```diff
@@ -210,25 +226,70 @@ mod tests {
 
   #[test]
   fn no_const_assign_invalid() {
-    assert_lint_err::<NoConstAssign>("const x = 0; x = 1;", 13);
-    assert_lint_err::<NoConstAssign>("const {a: x} = {a: 0}; x = 1;", 23);
-    assert_lint_err::<NoConstAssign>("const x = 0; ({x} = {x: 1});", 15);
-    assert_lint_err::<NoConstAssign>("const x = 0; ({a: x = 1} = {});", 14);
-    assert_lint_err::<NoConstAssign>("const x = 0; x += 1;", 13);
-    assert_lint_err::<NoConstAssign>("const x = 0; ++x;", 13);
-    assert_lint_err::<NoConstAssign>(
-      "const x = 0; function foo() { x = x + 1; }",
-      30,
-    );
-    assert_lint_err::<NoConstAssign>(
-      "const x = 0; function foo(a) { x = a; }",
-      31,
-    );
-    assert_lint_err::<NoConstAssign>("for (const i = 0; i < 10; ++i) {}", 26);
-    assert_lint_err::<NoConstAssign>(
-      "const x = 0; while (true) { x = x + 1; }",
-      28,
-    );
+    assert_lint_err! {
+      NoConstAssign,
+      r#"const x = 0; x = 1;"#: [
+      {
+        col: 13,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const {a: x} = {a: 0}; x = 1;"#: [
+      {
+        col:  23,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const x = 0; ({x} = {x: 1});"#: [
+      {
+        col:  15,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const x = 0; ({a: x = 1} = {});"#: [
+      {
+        col:  14,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const x = 0; x += 1;"#: [
+      {
+        col:  13,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const x = 0; ++x;"#: [
+      {
+        col:  13,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const x = 0; function foo() { x = x + 1; }"#: [
+      {
+        col:  30,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const x = 0; function foo(a) { x = a; }"#: [
+      {
+        col:  31,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"for (const i = 0; i < 10; ++i) {}"#: [
+      {
+        col:  26,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+      r#"const x = 0; while (true) { x = x + 1; }"#: [
+      {
+        col:  28,
+        message: NoConstantAssignMessage::Unexpected,
+        hint: NoConstantAssignHint::Remove,
+      }],
+    }
+
     assert_lint_err_on_line::<NoConstAssign>(
```

Comment:
Let's replace `assert_lint_err_on_line` and `assert_lint_err_n` with `assert_lint_err!` macro as well. 
An example of how to replace is: https://github.com/denoland/deno_lint/blob/main/src/rules/getter_return.rs#L648-L664

Reply:
Fixed in a68ef06

Reply:
great!